doc/api/esm.md

```diff
@@ -1220,6 +1177,50 @@ loaded from disk but before Node.js executes it; and so on for any `.coffee`,
 `.litcoffee` or `.coffee.md` files referenced via `import` statements of any
 loaded file.
 
+#### Overriding loader
+
+The previous two loaders defined `load` hooks. This is an example of a loader
+that does its work via the `resolve` hook. This loader reads an
+`overrides.json` file that specifies which specifiers to override to another
+URL.
+
+```js
+// overriding-loader.js
+import fs from 'node:fs/promises';
+
+const overrides = JSON.parse(await fs.readFile('overrides.json'));
+
+export async function resolve(specifier, context, nextResolve) {
+  if (specifier in overrides) {
+    return nextResolve(overrides[specifier], context);
+  }
+
+  return nextResolve(specifier, context);
+}
+```
+
+Let's assume we have these files:
+
+```js
+// main.js
+import 'a-module-to-override';
+```
+
+```json
+// overrides.json
+{
+  "a-module-to-override": "./module-override.js"
+}
```

Comment:
Should we use an import map instead?

Reply:
In essence, this _is_ an import map, but simplified. Do you think I should change it to conform to import map syntax? This has an advantage in that some people know about it, but:

1. Most people don't.
2. Supporting import maps fully is way to difficult and out of scope for this example
3. It will make the sample code more cumbersome.

I thought of using import maps, but the above three cons made me choose the simpler path. I can easily change if you think it makes more sense.

Reply:
If we’re using a JSON file, then I think it should be using the import map scheme. If someone builds a loader on top of this example, we are not making a service by using a syntax incompatible with import maps proposal when we could be using a subset of it. The fact that not everyone knows it is not an argument against it IMHO, because no one knows the syntax you came up with either. Also I don’t really see how it makes it more cumbersome, the only thing you need to do is to wrap it in an `"imports"` key, right? That seems pretty minimal to me. Supporting the full import maps proposal is out of the scope of this example, that I agree, let’s not try that. 

We can try using something that’s not a JSON file, but that would probably complicate things even further.

Reply:
Will do!